subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -0,0 +1,1343 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2018-2019 Foundries.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Uses some original concepts by:
+ *         Joakim Eriksson <joakime@sics.se>
+ *         Niclas Finne <nfi@sics.se>
+ *         Joel Hoglund <joel@sics.se>
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_registry
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <fcntl.h>
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#include <zephyr/net/tls_credentials.h>
+#endif
+#if defined(CONFIG_DNS_RESOLVER)
+#include <zephyr/net/dns_resolve.h>
+#endif
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_util.h"
+#if defined(CONFIG_LWM2M_RW_SENML_JSON_SUPPORT)
```

Comment:
My comment about independence from a single content format applies also to the remaining content format headers:
* #include "lwm2m_rw_senml_json.h"
* ...
* #include "lwm2m_rw_senml_cbor.h"

Those shall be removed too.